modules/library/data/src/main/java/org/geotools/data/store/ContentFeatureSource.java

```diff
@@ -387,12 +387,12 @@ public final ReferencedEnvelope getBounds(Query query) throws IOException {
             Diff diff = state.getDiff();
             
             // don't compute the bounds of the features that are modified or removed in the diff
-            Iterator it = diff.getModified().values().iterator();
+            Iterator it = diff.getModified().keySet().iterator();
             FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
             Set<FeatureId> modifiedFids = new HashSet<FeatureId>();
             while(it.hasNext()){
-                SimpleFeature feature = (SimpleFeature) it.next();
-                modifiedFids.add(ff.featureId(feature.getID()));
+                String featureId = (String)it.next();
+                modifiedFids.add(ff.featureId(featureId));
             }
             Id idFilter = ff.id(modifiedFids);
```

Comment:
This filter does not match what is described in the javadoc.

Expect it should be:

```
ff.not( ff.id( modifiedFids) )
```

I think we can construct a test case for this one.